src/Engine/ProtoCore/FFI/CLRDLLModule.cs

```diff
@@ -1342,13 +1331,22 @@ public FFIMethodAttributes(MethodBase method, Dictionary<MethodInfo, Attribute[]
             }
 
             Attribute[] atts;
+            attributes = new List<Attribute>();
             if (method is MethodInfo mInfo && getterAttributes != null && getterAttributes.TryGetValue(mInfo, out atts))
             {
-                attributes = atts;
+                attributes.AddRange(atts);
             }
             else
             {
-                attributes = method.GetCustomAttributes(false).Cast<Attribute>().ToArray();
+                attributes.AddRange(method.GetCustomAttributes(false).Cast<Attribute>());
+            }
+            if(method is MethodInfo minfo)
+            {
+                var returnTypeAtts = minfo.ReturnTypeCustomAttributes;
+                if (returnTypeAtts != null)
+                {
+                    attributes.AddRange(returnTypeAtts.GetCustomAttributes(false).Cast<Attribute>());
```

Comment:
so this seems to be compressing a bit of information - we seem to take the return type attribute and then assign it to a list of attributes for the entire node (method) - whats the purpose of making it an attribute of the `return` value if we just add it to the same list of attributes on the `FFIMethodAttributes` ?

Reply:
I guess we could make it an attribute on the entire method as we do for `AllowRankReduction`. Currently, the `ArbitraryDimensionArrayImport` attribute's usage is for parameters and _return_ values, and if we were to change, it, it would need to be applied to methods and properties as we do for `AllowRankReduction`, not sure if that is a bigger change and/or API break. I believe with the return qualifier it also makes it clearer that it's applied to the return value and not to something else on the method.